module-2/backend/week_one.md

```diff
@@ -5,28 +5,76 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
+GET, PUT, POST, PATCH, & DELETE
+GET is used to read resources (or pages) without changing them.
+POST is used create new resources, often on a web page further down "the tree".
+PUT is used to update an existing resource or page, or to create a new one when the user knows the "address" of the new resource.
+PATCH is used like PUT but in specific language (JSON or ?)
+DELETE is used to delete a resource or page.
+
 2. What is Sinatra?
+Sinatra is a sort of lite version of Rails.  Like Rails, it's a Ruby framework that enables the creation of websites.  But, compared to Rails, it's pretty simple and lacks much of Rails's more robust functions.
+
 4. What is MVC?
+Model-Controller-Views.  Its an architecture for creating interactive web applications.
+
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+So that others will more easily understand the functionality of our Ruby applications.
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+Instance variables...through ruby interpolation.
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+  @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
+name = "Mr. Ed"
+
 9. What's the purpose of ERB?
+It allows us to put Ruby code into HTML code and pages.
+
 10. Why do I need a development AND test database?
+So that we can test and develop code without corrupting crucial data that is needed for our production website.
+
 11. What's responsive design?
+It is design that adjusts to the size of the user's screen.
+
 12. What is CRUD and why is it important?
+CRUD is an acronym that stands for Create, Read, Update and Delete which are the four basic functions of a user interface.
+
 13. What does HTTP stand for? 
+Hyper-text Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+We can use regular interpolation or compact interpolation which is not often used and somewhat controversial.
+
 15. What's an ORM?
+Object Relational mapping, such as ActiveRecord which wraps SQL commands in code.
+
 16. What's the most commonly used ORM?
+Had to look this one up.  Evidently it's Hibernate which uses Java.

```

Comment:
Bad question - I should have specified the most commonly used ORM in Ruby - which is ActiveRecord 👍